x/nameservice/keeper/msg_server.go

```diff
@@ -39,7 +39,7 @@ func (m msgServer) SetRecord(c context.Context, msg *types.MsgSetRecord) (*types
 			types.EventTypeSetRecord,
 			sdk.NewAttribute(types.AttributeKeySigner, msg.GetSigner()),
 			sdk.NewAttribute(types.AttributeKeyBondID, msg.GetBondId()),
-			sdk.NewAttribute(types.AttributeKeyPayload, msg.Payload.String()),
+			sdk.NewAttribute(types.AttributeKeyPayload, msg.Payload.Record.Id),
```

Comment:
I like this change, seems better to emit only the ID, but want to double-check with Ashwin as to whether or not there was any reason to encode the entire payload into a tendermint event. Was there a specific motivation for this change?

Reply:
Now that the attributes are of type codectypes.Any, We cant unmarshal PayLoad to a string type. There can be a workaround to convert it to the specific Payload type based on attribute type, If that's necessary.